.circleci/config.yml

```diff
@@ -185,7 +191,63 @@ jobs:
       - install_solidus: { flags: "--sample=false --frontend=solidus_starter_frontend --with-authentication" }
       - test_page: { expected_text: "The only eCommerce platform you’ll ever need." }
 
+  test_solidus:
+    parameters:
+      database:
+        type: string
+        default: postgres
+      ruby:
+        type: string
+        default: '3.1'
+      rails:
+        type: string
+        default: "7.0"
+      paperclip:
+        type: boolean
+        default: true
+      legacy:
+        type: string
+        default: "0"
+    executor:
+      name: << parameters.database >>
+      ruby: << parameters.ruby >>
+    parallelism: &parallelism 3
+    environment:
+      DISABLE_ACTIVE_STORAGE: << parameters.paperclip >>
+      RAILS_VERSION: "~> << parameters.rails >>"
+      USE_LEGACY_EVENTS: << parameters.legacy >>
+      BUILDKITE_ANALYTICS_EXECUTION_NAME_PREFIX: "(<< parameters.ruby >>:<< parameters.rails >>:<< parameters.database >>:<< parameters.paperclip >>)"
+    steps:
+      - setup
+      - when:
+          condition:
+            not:
+              equal: [<<parameters.ruby>>, '2.5']
+          steps:
+            - libvips
+      - when:
+          condition:
+            equal: [<<parameters.ruby>>, '2.5']
+          steps:
+            - imagemagick
+      - test
+
 workflows:
   build:
     jobs:
       - solidus_installer
+
+      # Based on supported versions for the current Solidus release and recommended versions from
+      # https://www.fastruby.io/blog/ruby/rails/versions/compatibility-table.html.
+      - test_solidus:
+          name: &name "test-rails-<<matrix.rails>>-ruby-<<matrix.ruby>>-<<matrix.database>>-<<#matrix.paperclip>>paperclip<</matrix.paperclip>><<^matrix.paperclip>>activestorage<</matrix.paperclip>>"
+          matrix: { parameters: { rails: ['7.0'], ruby: ['3.0', '3.1'], database: ['mysql', 'sqlite', 'postgres'], paperclip: [true, false] } }
+      - test_solidus:
+          name: *name
+          matrix: { parameters: { rails: ['6.1'], ruby: ['2.7', '3.0'], database: ['sqlite'], paperclip: [false] } }
+      - test_solidus:
+          name: *name
+          matrix: { parameters: { rails: ['6.0'], ruby: ['2.6'], database: ['sqlite'], paperclip: [true] } }
+      - test_solidus:
+          name: *name
+          matrix: { parameters: { rails: ['5.2'], ruby: ['2.5'], database: ['sqlite'], paperclip: [true] } }
```

Comment:
@kennyadsl branch CI rules for v3.2 must be updated after this change to match the new job names 🎗️